backend/src/routes/game.py

```diff
@@ -53,7 +53,8 @@
             await check_timers(websocket, username, game)
             await game.push({"code": "players", "players": game.get_players_hp()})
         except (WebSocketDisconnect, ConnectionClosedError):
-            update_stats(game.players_data[username], username, False)
+            if username in game.players_data:
```

Comment:
It hides a serious bug in our code, imo you should leave a comment on that issue.

By the way, in Python we should use EAFP (easier to ask for forgiveness than permission) coding style. 
This `if` is a anti-pattern. You can read more [here](https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permssion_instead_of_forgiveness_when_working_with_files.html)